cli/silence_query.go

```diff
@@ -117,6 +123,11 @@ func query(cmd *cobra.Command, args []string) error {
 		if !expired && silence.EndsAt.Before(time.Now()) {
 			continue
 		}
+
+		if int64(within) > 0 && silence.EndsAt.After(time.Now().Add(within)) {
+			continue
```

Comment:
EndsAt is UTC on the AM, and calling time.Now() on amtool will use the user time zone, better to use UTC as well `time.Now().UTC().Add ...`